src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -172,10 +172,10 @@ xorGate i1 i2 o = [rise i1, rise i2] ~|~> rise o <> [fall i1, fall i2] ~|~> rise
                <> [rise i1, fall i2] ~|~> fall o <> [fall i1, rise i2] ~|~> fall o
 
 srLatch :: Eq a => a -> a -> a -> CircuitConcept a
-srLatch s r q = complexGate (Var s) (Var r) q
+srLatch s r q = complexGate (Var s) (Var r) q <> mutex s r
```

Comment:
Yes, that's the correct behaviour!

If you'd like to actually restrict inputs for the purposes of testing in the right environment, you can also append `me s r`. This will then restrict the inputs, perhaps giving more opportunities for synthesis tools.